src/Sulu/Bundle/AdminBundle/Resources/js/views/Datagrid/Datagrid.js

```diff
@@ -194,26 +260,19 @@ export default withToolbar(Datagrid, function() {
         }
         : undefined;
 
-    const items = [];
+    const items = this.toolbarActions
+        .map((toolbarAction) => toolbarAction.getToolbarItemConfig())
+        .filter((item) => item !== undefined);
 
     if (addRoute) {
-        items.push({
+        items.unshift({
```

Comment:
I think here we also have to discuss if we add toolbar buttons always the same way, as we do it for the forms. It might not be very intuitive if you only have added one toolbar action and two buttons appear